File: src/AddButton.java
import java.util.List;

/**
 * Button responsible for adding new books.
 * This code complies with the JMU honor code.
 * 
 * @author William Morris
 * @version 4/3/2024
 *
 */
public class AddButton extends EditButton {

	private static final String BUTTON_NAME = "Add";
	
	public AddButton(BookzController controller) {
		super(controller, BUTTON_NAME);
	}

	@Override
	void addListener() {
		button.addActionListener(arg0 -> {
			window = new EditWindow(BOOK_FIELDS, "Add Book", AddButton.this);
			window.setVisible(true);
		});
	}

	/**
	 * Called when user entry is complete.
	 */
	@Override
	void done() {
		List<String> fields = window.getReturnFields();
		controller.addBook(fields);
	}

}

File: src/main/kotlin/AddButton.kt

/**
 * Button responsible for adding new books.
 * This code complies with the JMU honor code.
 *
 * @author William Morris
 * @version 4/3/2024
 */
class AddButton(controller: BookzController) : EditButton(controller, BUTTON_NAME) {
    override fun addListener() {
        button.addActionListener {
            window = EditWindow(BOOK_FIELDS, "Add Book", this@AddButton)
            window?.setVisible(true)
        }
    }

    /**
     * Called when user entry is complete.
     */
    override fun done() {
        val fields = window!!.returnFields
        controller.addBook(fields)
    }

    companion object {
        private const val BUTTON_NAME = "Add"
    }
}
